### ISSUE TITLE
Unpublished items do not appear in menu "Parent item" select options

### ISSUE THREAD

**Issue by jenlampton** [backdrop/backdrop-issues#1558]
Unpublished items do not appear in menu "Parent item" select options

**Comment by jenlampton** [comment]
related https://github.com/backdrop/backdrop-issues/issues/1557

**Comment by BWPanda** [comment]
Should they be though...? If you add a published node's menu item as a child of an unpublished item, it won't appear in the menu (or it shouldn't). Won't that cause a WTF moment for users?

**Comment by indigoxela** [comment]
I agree with @BWPanda - IMO they shouldn't, because they will cause WTF moments (invisible menu items).

**Comment by klonos** [comment]
This makes it a workflow nightmare for people that need to create drafts though. They cannot add nodes under the proper parent, unless the parent in published.

**Comment by stpaultim** [comment]
I see the problem with both scenarios. 

If you are drafting a multi-page book (book content type) and are not yet ready to publish content, then it's REALLY weird that you can't make connections to other content. That is one of the main features of the book module, the ability to create sub-sections and parent/child relationships. 

Let's say I create chapter two of my book, but don't publish it. I think want to create chapter 2 section 1 of my book, but there is no way for me to create the relationship, UNTIL I am ready to publish. That feels broken to me as a content editor. 

![book-module](https://user-images.githubusercontent.com/3144571/104144320-5f08c500-5388-11eb-8779-c30d77eff6b2.png)

I also see the problem where a user might not realize this connection and add content to an unpublished node accidentally and NOT understand that the parent node is unpublished and that is why their new node is not showing up in the menu.

In my opinion, this is a problem that needs to be fixed, but the solution may be more difficult than simply showing unpublished items in a select list. Maybe unpublished items should be visible in the list, but maybe we need to make it clear that the item is unpublished. 

Maybe, when a user saves a child of an unpublished node, we provide a warning or status messages reminding them that this is a child of an unpublished node and will not show up in menus until the parent item is published.

**Comment by indigoxela** [comment]
> ..., but the solution may be more difficult than simply showing unpublished items in a select list. Maybe unpublished items should be visible in the list, but maybe we need to make it clear that the item is unpublished.

I think, that's the point. If we provide unpublished nodes in the list, it seems necessary to also provide the status of the node in some way. Just setting a message after saving might not be enough, though, as these messages ("... has been saved") won't get much attention.

**Comment by klonos** [comment]
I have just created #5970 as a wider issue (with how menu items of unpublished/scheduled content are shown in the site navigation and the menu links management forms).

This long-standing issue, which seems to be the equivalent of this issue here, was just fixed in D 10.1.x: [Allow menu items which link to unpublished nodes to be selected in the parent item selector](https://www.drupal.org/node/2807629). We had multiple tickets from customers in GovCMS complaining about this, and I imagine that there'd be more, but they either didn't get bothered to raise a ticket, or their dev partners might have already told them that it is not possible but coming soon.

Anyway, I believe that at the very least people with the "View any unpublished content", "View own unpublished content" or the "Bypass content access control" permissions should be allowed to see and use menus that belong to draft content. So I agree with @jenlampton that this is a bug and not expected behavior. If we are concerned about people selecting "invisible" menu items, then we can:
- indicate which items are disabled (link to unpublished content) in the select list
- add some help text along the lines of `If you select an unpublished menu item as parent, then this content item will only be able to be accessed by people with appropriate permissions.` (something like that)
